soccerdata/understat.py

```diff
@@ -0,0 +1,707 @@
+        ----------
+        force_cache : bool
+             By default no cached data is used for the current season.
+             If True, will force the use of cached data anyway.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        df_seasons = self.read_seasons()
+
+        stats = []
+        for (league, season), league_season in df_seasons.iterrows():
+            league_id = league_season["league_id"]
+            season_id = league_season["season_id"]
+            url = league_season["url"]
+
+            is_current_season = not self._is_complete(league, season)
+            no_cache = is_current_season and not force_cache
+
+            data = self._read_league_season(url, league_id, season_id, no_cache)
+
+            teams_data = data["teamsData"]
+            team_mapping = {}
+            for team in teams_data.values():
+                team_name = _as_str(team["title"])
+                team_id = _as_int(team["id"])
+                team_mapping[team_name] = team_id
+
+            players_data = data["playersData"]
+            for player in players_data:
+                player_team_name = player["team_title"]
+                if "," in player_team_name:  # pick first team if multiple teams are listed
+                    player_team_name = player_team_name.split(",")[0]
+                player_team_name = _as_str(player_team_name)
+                player_team_id = team_mapping[player_team_name]
+                stats.append(
+                    {
+                        "league": league,
+                        "league_id": league_id,
+                        "season": season,
+                        "season_id": season_id,
+                        "team": player_team_name,
+                        "team_id": player_team_id,
+                        "player": _as_str(player["player_name"]),
+                        "player_id": _as_int(player["id"]),
+                        "position": _as_str(player["position"]),
+                        "matches": _as_int(player["games"]),
+                        "minutes": _as_int(player["time"]),
+                        "goals": _as_int(player["goals"]),
+                        "xg": _as_float(player["xG"]),
+                        "np_goals": _as_int(player["npg"]),
+                        "np_xg": _as_float(player["npxG"]),
+                        "assists": _as_int(player["assists"]),
+                        "xa": _as_float(player["xA"]),
+                        "shots": _as_int(player["shots"]),
+                        "key_passes": _as_int(player["key_passes"]),
+                        "yellow_cards": _as_int(player["yellow_cards"]),
+                        "red_cards": _as_int(player["red_cards"]),
+                        "xg_chain": _as_float(player["xGChain"]),
+                        "xg_buildup": _as_float(player["xGBuildup"]),
+                    }
+                )
+
+        index = ["league", "season", "player"]
```

Comment:
Can you add "team" to the index for consistency with other methods and data sources?

Reply:
Fixed it myself :)

Reply:
Thanks!